### ISSUE TITLE
[BUG] "API_WHITELIST_IP" Not working after restart

### ISSUE THREAD

**Issue by 0FiRE0** [bunkerity/bunkerweb#640]
### What happened?

When the containers are created for the first time, everything works fine. However, sometimes, after restart, the main `bunkerweb` container starts to reject API Calls coming from the `scheduler` noting that the whole subnet is whitelisted.

### How to reproduce?

1- Run Bunkerweb normally and add some services using the UI (Everything works fine at this stage)
2- Restart the containers `docker compose restart`
3- The `scheduler` is not able to connect to the main `bunkerweb` container anymore, as it is not considered as part of the `API Whitelisted IP`
4- After that, all services configured already, including the UI, will stop working and start to through `404` errors

Note that this is an inconsistent behavior, sometimes it works after the restart (which did already with me), and sometimes not.

### Configuration file(s) (yaml or .env)

```YAML
version: "3.5"

services:
  mybunker:
    image: bunkerity/bunkerweb:1.5.2
    ports:
      - 80:8080
      - 443:8443
    environment:
      - SERVER_NAME=example.com # replace with your domain
      - MULTISITE=yes
      - AUTO_LETS_ENCRYPT=no
      - DISABLE_DEFAULT_SERVER=yes
      - USE_CLIENT_CACHE=yes
      - USE_GZIP=yes
      - API_WHITELIST_IP=127.0.0.0/8 10.20.30.0/24
      - example.com_USE_UI=yes
      - example.com_SERVE_FILES=no
      - example.com_USE_REVERSE_PROXY=yes
      - example.com_REVERSE_PROXY_URL=/admin # replace with another url
      - example.com_REVERSE_PROXY_HOST=http://bw-ui:7000
      - example.com_INTERCEPTED_ERROR_CODES=400 404 405 413 429 500 501 502 503 504
    labels:
      - "bunkerweb.INSTANCE"
    networks:
      - bw-universe
      - bw-services

  bw-scheduler:
    image: bunkerity/bunkerweb-scheduler:1.5.2
    depends_on:
      - mybunker
    environment:
      - DOCKER_HOST=tcp://bw-docker-proxy:2375
    volumes:
      - bw-data:/data
    networks:
      - bw-universe
      - bw-docker

  bw-ui:
    image: bunkerity/bunkerweb-ui:1.5.2
    depends_on:
      - bw-docker-proxy
    environment:
      - DOCKER_HOST=tcp://bw-docker-proxy:2375
      - ADMIN_USERNAME=admin
      - ADMIN_PASSWORD=123456!Aa # replace with a stronger password
    volumes:
      - bw-data:/data
    networks:
      - bw-universe
      - bw-docker

  bw-docker-proxy:
    image: tecnativa/docker-socket-proxy:0.1
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
    environment:
      - CONTAINERS=1
      - LOG_LEVEL=warning
    networks:
      - bw-docker

volumes:
  bw-data:

networks:
  bw-universe:
    name: bw-universe
    ipam:
      driver: default
      config:
        - subnet: 10.20.30.0/24
  bw-services:
  bw-docker:
```

### Relevant log output

```shell
bunkerweb-bw-scheduler-1     | [2023-09-14 18:43:38] - API - ❌ - Can't send API request to http://bunkerweb-mybunker-1:5000/confs : Request failed: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
bunkerweb-bw-scheduler-1     | [2023-09-14 18:43:38] - SCHEDULER - ❌ - Sending nginx configs failed, configuration will not work as expected...
bunkerweb-mybunker-1         | 2023/09/14 18:43:38 [warn] 45#45: *16 [API] can't validate access from IP 10.20.30.3 : IP is not in API_WHITELIST_IP, client: 10.20.30.3, server: bwapi, request: "POST /confs HTTP/1.1", host: "bwapi"
bunkerweb-mybunker-1         | bwapi 10.20.30.3 - - [14/Sep/2023:18:43:38 +0000] "POST /confs HTTP/1.1" 444 0 "-" "bwapi"
bunkerweb-mybunker-1         | 2023/09/14 18:43:39 [warn] 46#46: *17 [API] can't validate access from IP 10.20.30.3 : IP is not in API_WHITELIST_IP, client: 10.20.30.3, server: bwapi, request: "POST /cache HTTP/1.1", host: "bwapi"
bunkerweb-mybunker-1         | bwapi 10.20.30.3 - - [14/Sep/2023:18:43:39 +0000] "POST /cache HTTP/1.1" 444 0 "-" "bwapi"
bunkerweb-bw-scheduler-1     | [2023-09-14 18:43:39] - API - ❌ - Can't send API request to http://bunkerweb-mybunker-1:5000/cache : Request failed: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
bunkerweb-bw-scheduler-1     | [2023-09-14 18:43:39] - SCHEDULER - ❌ - Error while sending /var/cache/bunkerweb folder
bunkerweb-mybunker-1         | 2023/09/14 18:43:39 [warn] 47#47: *18 [API] can't validate access from IP 10.20.30.3 : IP is not in API_WHITELIST_IP, client: 10.20.30.3, server: bwapi, request: "POST /reload HTTP/1.1", host: "bwapi"
bunkerweb-mybunker-1         | bwapi 10.20.30.3 - - [14/Sep/2023:18:43:39 +0000] "POST /reload HTTP/1.1" 444 0 "-" "bwapi"
bunkerweb-bw-scheduler-1     | [2023-09-14 18:43:39] - API - ❌ - Can't send API request to http://bunkerweb-mybunker-1:5000/reload : Request failed: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
bunkerweb-bw-scheduler-1     | [2023-09-14 18:43:39] - SCHEDULER - ❌ - Error while reloading nginx
bunkerweb-bw-scheduler-1     | [2023-09-14 18:43:39] - SCHEDULER - ℹ️  - Executing job scheduler ...
```

### BunkerWeb version

1.5.2

### What integration are you using?

Docker

### Removed private data

- [X] I have removed all private data from the configuration file and the logs

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by 0FiRE0** [comment]
The only thing which resolved the issue, is to take down all the containers `docker compose down` and bring them up again `docker compose up -d`.
_Note that the `docker-compose.yml` file hasn't been changed at all from the first run._

Below is the successful log after doing that:

```
--- SOME LOGS ---

bunkerweb-mybunker-1         | 2023/09/14 18:55:53 [notice] 56#56: *10 [API] validated access from IP 10.20.30.4, client: 10.20.30.4, server: bwapi, request: "POST /confs HTTP/1.1", host: "bwapi"
bunkerweb-mybunker-1         | 2023/09/14 18:55:53 [notice] 56#56: signal 17 (SIGCHLD) received from 61
bunkerweb-mybunker-1         | 2023/09/14 18:55:53 [notice] 56#56: signal 17 (SIGCHLD) received from 62
bunkerweb-mybunker-1         | 2023/09/14 18:55:53 [notice] 56#56: *10 [API] successful call from 10.20.30.4 on /confs : saved data at /etc/nginx, client: 10.20.30.4, server: bwapi, request: "POST /confs HTTP/1.1", host: "bwapi"
bunkerweb-mybunker-1         | bwapi 10.20.30.4 - - [14/Sep/2023:18:55:53 +0000] "POST /confs HTTP/1.1" 200 67 "-" "bwapi"
bunkerweb-bw-scheduler-1     | [2023-09-14 18:55:53] - API - ℹ️  - Successfully sent API request to http://bunkerweb-mybunker-1:5000/confs
bunkerweb-mybunker-1         | 2023/09/14 18:55:54 [notice] 57#57: *11 [API] validated access from IP 10.20.30.4, client: 10.20.30.4, server: bwapi, request: "POST /cache HTTP/1.1", host: "bwapi"
bunkerweb-mybunker-1         | 2023/09/14 18:55:54 [notice] 57#57: signal 17 (SIGCHLD) received from 64
bunkerweb-mybunker-1         | 2023/09/14 18:55:54 [notice] 57#57: signal 17 (SIGCHLD) received from 65
bunkerweb-mybunker-1         | 2023/09/14 18:55:54 [notice] 57#57: *11 [API] successful call from 10.20.30.4 on /cache : saved data at /var/cache/bunkerweb, client: 10.20.30.4, server: bwapi, request: "POST /cache HTTP/1.1", host: "bwapi"
bunkerweb-mybunker-1         | bwapi 10.20.30.4 - - [14/Sep/2023:18:55:54 +0000] "POST /cache HTTP/1.1" 200 78 "-" "bwapi"
bunkerweb-bw-scheduler-1     | [2023-09-14 18:55:54] - API - ℹ️  - Successfully sent API request to http://bunkerweb-mybunker-1:5000/cache
bunkerweb-bw-scheduler-1     | [2023-09-14 18:55:54] - SCHEDULER - ℹ️  - Successfully sent /var/cache/bunkerweb folder
bunkerweb-mybunker-1         | 2023/09/14 18:55:54 [notice] 55#55: *12 [API] validated access from IP 10.20.30.4, client: 10.20.30.4, server: bwapi, request: "POST /reload HTTP/1.1", host: "bwapi"
bunkerweb-mybunker-1         | 2023/09/14 18:55:54 [notice] 55#55: *12 [API] successful call from 10.20.30.4 on /reload : reload successful, client: 10.20.30.4, server: bwapi, request: "POST /reload HTTP/1.1", host: "bwapi"
bunkerweb-mybunker-1         | bwapi 10.20.30.4 - - [14/Sep/2023:18:55:54 +0000] "POST /reload HTTP/1.1" 200 58 "-" "bwapi"
bunkerweb-bw-scheduler-1     | [2023-09-14 18:55:54] - API - ℹ️  - Successfully sent API request to http://bunkerweb-mybunker-1:5000/reload
bunkerweb-bw-scheduler-1     | [2023-09-14 18:55:54] - SCHEDULER - ℹ️  - Successfully reloaded nginx
bunkerweb-mybunker-1         | 2023/09/14 18:55:54 [notice] 51#51: signal 1 (SIGHUP) received from 55, reconfiguring
bunkerweb-bw-scheduler-1     | [2023-09-14 18:55:54] - SCHEDULER - ℹ️  - Executing job scheduler ...
bunkerweb-mybunker-1         | 2023/09/14 18:55:54 [notice] 51#51: reconfiguring

--- SOME LOGS ---
```

**Comment by TheophileDiot** [comment]
Hi @0FiRE0, thank you for this issue, I'll investigate and let you know 😁

**Comment by TheophileDiot** [comment]
@0FiRE0 Thank you for finding out this, I never use the `docker compose restart` command 😅

**Comment by fl0ppy-d1sk** [comment]
Hello @0FiRE0,

It should be fixed in the last 1.5.3 release. Don't hesitate to open a new issue if needed.
